### PR TITLE
Unique storage account

### PR DESCRIPTION
**Pull request by clguimanMSFT**
The storage account name has to be globally unique, not just per resource group name.
I've added the subscription id as part of the hash generation to ensure that.

Reported bug: https://issues.jenkins-ci.org/browse/JENKINS-40288

### CODE REVIEW

**Comment by rtyler** [comment]
@clguimanMSFT FWIW, referencing the JIRA (like [JENKINS-1234]) in the commit message should be all that's needed to link the pull request in the JIRA ticket

**Comment by arroyc** [comment]
can we also have an inline comment/help for the storage account field addressing this issue?

**Comment by clguimanMSFT** [comment]
@arroyc The feedback should come when you click verify template (that doesn't work either, but it will get fixed ASAP :) )

**Comment by clguimanMSFT** [comment]
image

File: src/main/webapp/help-storageAccountName.html [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <div>
-	Specify a storage account name. Leave blank to create a new storage account using the default name &quot;jenkinsarmst&quot;.<br />
-    <br />
+	Specify a storage account name. Leave blank to create a new storage account.<br />
+        <br/>
+        <b>The storage account name has to be unique!</b>
```
Comment (arroyc) [inline src/main/webapp/help-storageAccountName.html]: has to be globally unique across azure.

**Pull request merged**